src/constraints.jl

```diff
@@ -727,3 +727,313 @@ function list_of_constraint_types(model::Model)
+            end
+            if S <: MOI.GreaterThan
+                push!(var_lower, constr_obj.set.lower)
+                push!(var_upper, Inf)
+            end
+        end
+    end
+
+    A = sparse(row_i, col_j, coeffs, m, n)
+    return A, var_lower, var_upper, affine_constraints
+end
+"""
+Returns the optimal primal values for the problem in standard form, c.f. `_std_matrix(model)`.
+"""
+function _std_primal_solution(model::Model, constraints::Vector{ConstraintRef})
+    vars = all_variables(model)
+    x = [value.(vars); value.(constraints)]
+    return x
+end
+
+
+"""
+Returns the basis status of all variables of the problem in standard form, c.f. `_std_matrix(model)`.
+"""
+function _std_basis_status(model::Model)::Vector{MOI.BasisStatusCode}
+    con_types = list_of_constraint_types(model)
+    vars = all_variables(model)
+    var2idx = Dict{VariableRef, Int64}()
+    for (j, var) in pairs(vars)
+        var2idx[var] = j
+    end
+    basis_status = fill(MOI.BASIC, length(vars))
+
+    for (F, S) in con_types
+        constraints = all_constraints(model, F, S)
+        if F <: VariableRef
+            for (i, constr) in pairs(constraints)
+                constr_obj = constraint_object(constr)
+                j = var2idx[constr_obj.func]
+                basis_status[j] = MOI.get(model, MOI.ConstraintBasisStatus(), constr)
+                if S <: MOI.LessThan && basis_status[j] == MOI.NONBASIC
+                    basis_status[j] = MOI.NONBASIC_AT_UPPER
+                elseif S <: MOI.GreaterThan && basis_status[j] == MOI.NONBASIC
+                    basis_status[j] = MOI.NONBASIC_AT_LOWER
+                end
+            end
+            #issue 1892 makes is hard to know how to handle variable constraints, atm. assume unique variable constraints.
+            continue
+        end
+        for constr in constraints
+            con_basis_status = MOI.get(model, MOI.ConstraintBasisStatus(), constr)
+            if S <: MOI.LessThan && con_basis_status == MOI.NONBASIC
+                con_basis_status = MOI.NONBASIC_AT_UPPER
+            elseif S <: MOI.GreaterThan && con_basis_status == MOI.NONBASIC
+                con_basis_status = MOI.NONBASIC_AT_LOWER
+            end
+            push!(basis_status, con_basis_status)
+        end
+    end
+    return basis_status
+end
+
+function perturbation_range_of_feasibility(constraint::ConstraintRef{Model, _MOICON{F, S}}
+                      ) where {S <: Union{MOI.LessThan, MOI.GreaterThan, MOI.EqualTo}, T, F <: Union{MOI.ScalarAffineFunction{T}, MOI.SingleVariable}}
+    model = constraint.model
+    if termination_status(model) != MOI.OPTIMAL
+        error("The range of feasibility is not available because the current solution "*
+              "is not optimal.")
+    end
+
+    # This throw the error: ... do not support accessing ... ConstraintBasisStatus(), if its not implemented by the solver.
+    con_status = MOI.get(model, MOI.ConstraintBasisStatus(), constraint)
```

Comment:
Can we catch and throw a more informative error? This function will like be used by new users who don't know that `ConstraintBasisStatus` means.